### PR TITLE
Gabrielcsapo/typescript conversion

### PR DESCRIPTION
**Pull request by 5GameMaker**

### CODE REVIEW

**Comment by gabrielcsapo** [comment]
I have a working branch with changes let me create a PR I have been working on it a bunch and maybe we can merge

File: src/util.ts [hunk 1]
```diff
@@ -147,7 +147,7 @@ export function infoResponse(
  */
 export function parseGitName(repo: string): string {
   const locationOfGit = repo.lastIndexOf('.git');
-  return repo.substr(0, locationOfGit > 0 ? locationOfGit : repo.length);
```
Comment (gabrielcsapo) [inline src/util.ts:150]: why do we want to use slice here?

File: package.json [hunk 2]
```diff
@@ -18,15 +22,16 @@
...
     "node": ">= 14"
   },
   "scripts": {
     "lint": "eslint src/*",
     "test": "jest",
-    "build": "tsc",
-    "coverage": "jest --coverage"
+    "build": "echo \"\" > ./build-complete.meta && tsc",
+    "coverage": "jest --coverage",
+    "install": "( [ -f ./git-package.meta ] && [ ! -f ./build-complete.meta ] && echo \"\" > ./build-complete.meta && npm install . && tsc ) || echo \"\" > ./build-complete.meta"
```
Comment (gabrielcsapo) [inline package.json:34]: what does this do?

**Comment by gabrielcsapo** [comment]
Thank you for this! I just had a few questions!

File: src/util.ts (hunk 1 above)
Comment (5GameMaker) [reply src/util.ts:150]: `String.substr` is deprecated, so I replaced it with `slice` instead as with 0 being the first argument they're doing the same thing

File: package.json (hunk 2 above)
Comment (5GameMaker) [reply package.json:34]: For some reason npm does not build any package installed from git, so I added this cos it was too annoying (however I forgot to delete `dist` while testing, so I noticed that it doesn't work only when I committed this)

**Comment by gabrielcsapo** [comment]
Thank you @5GameMaker, merged and going to publish a beta.

**Pull request merged**